source/assets/js/playground/autocomplete.ts

```diff
@@ -0,0 +1,245 @@
+import {
+  CompletionContext,
+  CompletionResult,
+  CompletionSource,
+} from '@codemirror/autocomplete';
+import {sassCompletionSource} from '@codemirror/lang-sass';
+import {syntaxTree} from '@codemirror/language';
+import {EditorState} from '@codemirror/state';
+import moduleMetadata from './module-metadata';
+
+// The validFor identifier, from @codemirror/lang-css. After an initial set of
+// possible completions are returned from a completion soruce, the matched set
+// narrows as the user types as long as it matches this identifier. Once it no
+// longer matches, the completion sources are checked again.
+// https://codemirror.net/docs/ref/#autocomplete.CompletionResult.validFor
+const identifier = /^(\w[\w-]*|-\w[\w-]*|)$/;
+
+// Sass-specific at rules only. CSS at rules should be added to `@codemirror/lang-css`.
+const atRuleKeywords = [
+  'use',
+  'forward',
+  'import',
+  'mixin',
+  'include',
+  'function',
+  'extend',
+  'error',
+  'warn',
+  'debug',
+  'at-root',
+  'if',
+  'else',
+  'each',
+  'for',
+  'while',
+];
+
+// CompletionResult options for Sass at rules, for example `@use`.
+const atRuleOptions = Object.freeze(
+  atRuleKeywords.map(keyword => ({
+    label: `@${keyword} `,
+    type: 'keyword',
+    validFor: identifier,
+  }))
+);
+
+// Completions for Sass at rules
+function atRuleCompletion(context: CompletionContext): CompletionResult | null {
+  const atRule = context.matchBefore(/@\w*/);
+  if (!atRule) return null;
+  if (atRule.from === atRule.to && !context.explicit) return null;
+  return {
+    from: atRule.from,
+    to: atRule.to,
+    options: atRuleOptions,
+    validFor: identifier,
+  };
+}
+
+// A list of all the Sass built in modules.
+const moduleNames = moduleMetadata.map(mod => mod.name);
+type ModuleName = (typeof moduleNames)[number];
+
+// Matches any of the built in Sass modules.
```

Comment:
```suggestion
// Matches an identifier namespaced within any of the built in Sass modules.
```

Reply:
Fixed in https://github.com/sass/sass-site/pull/1166/commits/7c4ef88299d155c3d4db97b36d502220278162a6